community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/ast/rewriters/isolateAggregation.scala

```diff
@@ -92,6 +73,92 @@ case object isolateAggregation extends Rewriter {
       q.copy(clauses = newClauses)(q.position)
   }
 
+  private def createRewriterFor(withReturnItems: Set[ReturnItem]): Rewriter = {
+    def inner: Rewriter =
+      Rewriter.lift {
+        case original: Expression =>
+          val rewrittenExpression = withReturnItems.collectFirst {
+            case item@AliasedReturnItem(expression, variable) if original == expression =>
+              item.alias.get.copyId
+          }
+          rewrittenExpression getOrElse original
+      }
+
+    /*
+    Instead of using topDown, we do it manually here, because we don't want to rewrite the return aliases,
+    only the expressions
+    */
+    new Rewriter {
+
+      override def apply(that: AnyRef): AnyRef = {
+        val initialStack = mutable.ArrayStack((List(that), new mutable.MutableList[AnyRef]()))
+        val result = rec(initialStack)
+        assert(result.size == 1)
+        result.head
+      }
+
+      @tailrec
+      def rec(stack: mutable.ArrayStack[(List[AnyRef], mutable.MutableList[AnyRef])]): mutable.MutableList[AnyRef] = {
```

Comment:
It may because I have looked at too much java code lately but could we give this a little more descriptive name, it takes me at least some work to figure out what this method does.